redfish-core/lib/cable.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+#include <boost/container/flat_map.hpp>
+#include <utils/json_utils.hpp>
+
+namespace redfish
+{
+using ManagedPropertyType =
+    boost::container::flat_map<std::string, std::variant<std::string, double>>;
+
+/**
+ * @brief Get Cable properties
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       resp        A map to hold Cable properties
+ * @param[in]       cableId     Cable Id
+ */
+inline void
+    getCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       ManagedPropertyType& resp, const std::string& cableId)
+{
+    for (auto& propertyMap : resp)
+    {
+        if (propertyMap.first == "CableTypeDescription")
+        {
+            std::string* cableTypeDescription =
+                std::get_if<std::string>(&propertyMap.second);
+            if (cableTypeDescription != nullptr)
+            {
+                asyncResp->res.jsonValue["CableType"] = *cableTypeDescription;
+            }
+        }
+        else if (propertyMap.first == "Length")
+        {
+            double* cableLength = std::get_if<double>(&propertyMap.second);
+            if (cableLength != nullptr)
```

Comment:
again typically it is  
if (cableLength == nullptr) 
{
internalError; 
return

Reply:
Done